foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -0,0 +1,260 @@
+import time
+from typing import Callable, List
+from threading import Thread
+from functools import wraps
+from retrying import retry
+from datetime import timedelta
+from django.utils import timezone
+from django.db import transaction
+from data_refinery_models.models import (
+    WorkerJob,
+    DownloaderJob,
+    ProcessorJob
+)
+from data_refinery_foreman.surveyor.message_queue import app
+
+# Import and set logger
+import logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# Maximum number of retries, so the number of attempts will be one
+# greater than this because of the first attempt
+MAX_NUM_RETRIES = 2
+
+# For now it seems like no jobs should take longer than a day to be
+# either picked up or run.
+MAX_DOWNLOADER_RUN_TIME = timedelta(days=1)
+MAX_PROCESSOR_RUN_TIME = timedelta(days=1)
+MAX_QUEUE_TIME = timedelta(days=1)
+
+# To prevent excessive spinning loop no more than once every 10
+# seconds.
```

Comment:
Because if it's all on one line then the line length is exactly 72 characters long, where PEP8 specifies that all comments and docstrings should be shorter than 72...  It's not my favorite style rule but I've gotten in the habit of just hitting `M-q` and letting emacs decide how to space it out (since it follows the guide exactly).